### DISCUSSION TITLE
Feature Request: macOS Support

### DISCUSSION THREAD

**Discussion by Saadnajmi** [software-mansion/react-native-screens#2589] [Ideas]
Hello! I noticed this library doesn't have macOS support yet. I think there are some parallels (Header -> Window Header, Screen -> An NSView presented from a sidebar or something). Adding support for this would also be a great boon to helping more React Native apps be cross platform, since this library is heavily used :)

I came across this because there are a few test apps of other libraries (react-native-skia) for example that use RNScreens, so to port that library's test app on macOS, I have gotten stuck / forked the code.

I'd be willing to also submit a PR myself, though it might take me a while. The last I checked, there was _a lot_ of UIKit code in this repo, so simple ifdefs may not suffice. Maybe a small limited implementation just to unblock support? I'll have to investigate.

**Comment by kkafar** [comment]
@Saadnajmi We dropped idea of macOS support some time ago as the macOS does not natively support stack-navigation paradigm and using `screens` there would not effectively bring any advantages to using plain views. I can see some value of making `screens` work there anyway, as this would allow easier app-porting, however I've never given it much more thought. I'll be happy, however to discuss & guide in case you find time to open a PR and work on this feature.

**Reply by Saadnajmi** [reply]
Makes sense! I also realize you have the `enableScreens` bool you export, so perhaps I can be unblocked by just disabling it for macOS
